src/main/java/com/userservice/controller/UserController.java

```diff
@@ -40,21 +44,22 @@ public UserController(UserService userService) {
      * @return ResponseEntity<User>
      */
     @GetMapping("/users/{id}")
-    public ResponseEntity<User> getUserDetailsById(@PathVariable String id) {
-        log.info("Entering UserController.getUserDetailsById with parameter id {}.", id);
-        return ResponseEntity.ok().body(userService.findUserById(id));
+    public ResponseEntity<User> getUserDetailsById(@PathVariable String id, HttpServletRequest request) {
+        log.info("Entering UserController.getUserDetailsById with parameter id {}, userName {}", id, request.getAttribute("userName"));
+        if (request.getAttribute("id") != null)
+            return ResponseEntity.ok().body(userService.findUserById((id), request.getAttribute("id").toString()));
```

Comment:
casting to `String` should be safer, like you are doing below

Reply:
Done.